app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(name=person["name"], age=person["age"])
+                   for person in people]
+    for person in people:
+        if person.get("husband"):
+            main_person = Person.people[person["name"]]
+            husband_person = Person.people[person["husband"]]
+            main_person.husband = husband_person
+        elif person.get("wife"):
```

Comment:
You can avoid nested `if` statements by using `and` operator. For example, you can check if `person` has a 'wife' and 'name' at once. This is a code style issue. Here is an example of how you can do it: `if person.get("wife") and person.get("name"):`.